### DISCUSSION TITLE
How do I add pyinstaller to path

### DISCUSSION THREAD

**Discussion by DannyBrock123** [pyinstaller/pyinstaller#7536] [Help]
Hi, I just got pyinstaller and its having some problems, it didn't add itself to the system path file. Can you please help me how to add it to path. Cmd is not finding the application and I need helping adding it to path

**Comment by bwoodsend** [comment]
The same way you add any application to `PATH` (which is an environment variable - not a file). Find `pyinstaller.exe` (pip should have given you a warning telling you where it is on install), then copy the directory it's in. Then find the *edit system environment variables* setting on your PC, find `PATH` and paste the directory there. I'm rather reluctant to tell you this - you're going to need be familiar with much worse than `PATH` to get a functional application out of PyInstaller.

**Reply by DannyBrock123** [reply]
Where is pyinstaller.exe found?

**Reply by rokm** [reply]
In your python environment's scripts directory. If you have system wide installation, this might be somewhere in user's AppData directory, and hard to track down. But when you installed PyInstaller, `pip` likely gave you a warning about it, and that warning contained the path to scripts directory.

So if you have no idea how to proceed, it might be easier to just invoke PyInstaller as a module (`python -m  PyInstaller` instead of `pyinstaller`). Or, create a virtual environment and use that instead (since activating a venv will put its scripts directory into `PATH`).

**Reply by DannyBrock123** [reply]
I was able to track down the pyinstaller.exe but when I copied it to path and tried it, It did not work. I'm I doing something wrong? Can you please add me on discord Markos#5515 so we can have direct chat and message. I need a lot of please. Thanks

**Reply by bwoodsend** [reply]
We're not a support service. Like I said, if this is out of your depth then you shouldn't be using PyInstaller.

**Reply by Haxxed** [reply]
Then who is d1ck H3ad, Ur the founder's. We need help with ur sh1tty stuff, why not just make the whole file a .exe so its simple to use and not all this python stuff, bad code and bad support.

**Reply by dacointrapper** [reply]
i really have never understood the whole nose in the air, you're better than me, sort of posture the veteran linux/programming/devops individuals take when someone is just legit asking for help.  All that condescending behind the screen bully talk goes right out the window if I ask you in person....

**Reply by bwoodsend** [reply]
> i really have never understood the whole nose in the air, you're better than me

Because it's not *nose in the air, you're better than me* posturing. Telling someone that their out of their depth doesn't imply that they're inferior. It just means that if I answer their question, they will still be stuck (as evident by the fact that we've already answered this question and people are still moaning at us).

>  All that condescending behind the screen bully talk goes right out the window if I ask you in person....

If you asked me a question like this in person I'd still tell you to look it up. Online tutorials will explain `PATH` better than me.

And to be clear, by *help*, you're really asking for hours of free personalised consultancy services over a matter that has nothing to do with the project we maintain (you'd get the same error with pytest or flake8) and a quick *how to add a program to PATH* google search would answer. We have millions of people using this software and we do not have millions of several hours to give each user 1:1 (possibly private so that no one else may benefit from it) support that you're implying that you're *entitled* to. The only fair thing we can do is to limit ourselves to the people with legitimate PyInstaller-specifc issues that can't be solved by a bit of googling.

Honestly, I used to feel the same way as you. But it only took a few weeks of people take advantage of that and treating me like a chatbot whose time is worthless to end up with my so-called *nose in the air* stance.

**Comment by rhit-pappasad** [comment]
> We're not a support service. Like I said, if this is out of your depth then you shouldn't be using PyInstaller.

Yes you literally are a support service.

**Comment by thiccoyaki** [comment]
Greetings fellow python coders and confused children of Pyinstaller.

If you have installed Python via Ms Store, your python install paths might be different
It might be in _C:\Users\[USERNAME]\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0_ or something similar.

This means your .exes that were installed via pip install might also be in a different path.

Your pyInstaller's files will be stored in _C:\Users\[USERNAME]\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages_

BUT, the exe will be stored in 
C:\Users\[USERNAME]\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\Scripts

The latter is what you should add to path.

Let me know if this helped!
